### PR TITLE
Document data synchronization via SyncStoredX natives

### PR DESCRIPTION
**Pull request by lep**
Trying to document the data synchronisation process using SyncStoredX natives.
I think I've got the process right but I would like to hear your opinion @WaterKnight @Luashine @cipherxof @Frotty 
Also I don't know how those new `BlzSendSyncData` stuff fits into this.

### CODE REVIEW

**Comment by lep** [comment]
Other questions: what does `TriggerSyncStart` do? What about `SyncSelections`?

**Comment by Luashine** [comment]
The reason I opened #63 (as you can guess) is that I don't know but wanted to save the links I found and hoped that someone else knew. I didn't read them yet :) Of all of us I've probably spent the least time reading guides and whatnot.

If we wanted to ask about these for real (outside of user-side knowledge) then we should consider pinging Unryze and MindWorX. 

iirc `TriggerSyncStart` is used in FileIO and is actually a "sync finished" event, `SyncSelections` is currently selected units for all players.

File: gamecache.j [hunk 1]
```diff
@@ -28,14 +28,63 @@ native  StoreString						takes gamecache cache, string missionKey, string key, s
 
 
 
+/**
+Synchronizes the value stored in the `gamecache` under the mission key and key.
+Calling this function sends a sync packet from each player in the calling
+context (citation needed), that is everybody if called simply as is.
```
Comment (Luashine) [inline gamecache.j:34]: > Calling this function sends a sync packet from each player in the calling
> context (citation needed), _that is everybody if called simply as is._

Worded ambiguously. 

> context (citation needed). That is, a simple call sends a packet from everyone to everyone.
> If you want only one player be the source of the sync, execute it locally with `GetLocalPlayer`.

File: gamecache.j [hunk 2]
```diff
@@ -28,14 +28,63 @@ native  StoreString						takes gamecache cache, string missionKey, string key, s
...
+The game then picks the first packet arrived (at the host). Often (but not
+lways) that is the packet coming from the game host.
```
Comment (Luashine) [inline gamecache.j:36]: a missing in `lways`

File: gamecache.j [hunk 3]
```diff
@@ -28,14 +28,63 @@ native  StoreString						takes gamecache cache, string missionKey, string key, s
...
+
+More interesting perhaps is the use to synchronize local data (like a players
```
Comment (Luashine) [inline gamecache.j:38]: player**'s** apostrophe

File: gamecache.j [hunk 4]
```diff
@@ -28,14 +28,63 @@ native  StoreString						takes gamecache cache, string missionKey, string key, s
...
+`TriggerRegisterPlayerUnitEvent(trig, p, EVENT_PLAYER_UNIT_SELECTED, null)`.
+Now once the last player has sent their selection event you have synchronized
+your data.
+
+This is a very high-level overview and the process has many edges to look out
+for, so it's probably a good idea to use an already made system like
+[this one](https://www.hiveworkshop.com/threads/sync-game-cache.279148/).
+
+@note Calling multiple `SyncStoredX` in a row will keep their order in the
+syncing process.
```
Comment (Luashine) [inline gamecache.j:67]: Add: `i.e. first sync will be received first (FIFO).` for clarity

File: trigger.j [hunk 5]
```diff
@@ -226,4 +226,8 @@ native TriggerExecuteWait   takes trigger whichTrigger returns nothing
 
 native TriggerSyncStart     takes nothing returns nothing
 
+/**
+Waits until all currently running `SynctedStoredX` (like `SyncStoredInteger`)
```
Comment (Luashine) [inline trigger.j:230]: SynctedStoredX -> SyncStoredX

**Review by Luashine — CHANGES_REQUESTED** [review]

File: gamecache.j (hunk 4 above)
Comment (WaterKnight) [reply gamecache.j]: Is this guaranteed even if the network packets are in disorder? Are the sync actions numbered?

Comment (lep) [reply gamecache.j]: I don't know but I think everyone is assuming it is.

**Pull request merged**